core/base/test/CompressionTests.cxx

```diff
@@ -1,38 +0,0 @@
-#include "TInterpreter.h"
```

Comment:
Can this end up somewhere else, or do we really not need it anymore? Could we e.g. assert that libCore exports no zlib symbols?

Reply:
I checked that the symbols are there with a `t`, so we should not need the test anymore. The symbols cannot be found because they are hidden.

Reply:
Sure, that's the case after this PR. But do you think it could be useful to assert that this is the case also after future changes? I.e. to have a test that actually checks that? (Similar in spirit to what this test was doing?)

Reply:
What do you mean by "this is the case"? After this PR the symbols will be hidden, so the user at the prompt will always get the system version of zlib, which does not need to match the one bundled in ROOT. We could, however, check that no zlib symbols are exported by libCore if that's what you mean.

Reply:
Clarified verbally - summary: zlib shouldn't be autoloaded, so for builtin zlib no zlib symbols should be found. And it would be good to assert that with a test.